packages/graph/hash_graph/lib/graph/src/store/crud.rs

```diff
@@ -0,0 +1,22 @@
+//! Store interface for CRUD operations.
+//!
+//! The traits defined in this module are used in [`Store`] to create, read, update, and delete
+//! entries. They form a unified access to the [`Store`], so it's possible to add operations to the
+//! [`Store`] without changing the [`Store`] implementation.
+
+use async_trait::async_trait;
+use error_stack::Result;
+
+use crate::store::{QueryError, Store};
+
+/// Read access to a [`Store`].
+#[async_trait]
+pub trait Read<I: Sync, T>: Store {
+    /// Output returned when getting the value by the index `I`.
+    type Output;
+
+    /// Returns a value from the [`Store`] identified by `index`.
+    async fn get(&self, index: &I) -> Result<Self::Output, QueryError>;
```

Comment:
I find the name `index` slightly confusing. Do you think we can find another name?

I was thinking `key` or `identifier` would be more intuitive, but `Index` felt a little strange. What are your thoughts?

Reply:
What do you think is confusing about `index`? The term is borrowed from `std` where the trait on `I` is called `SliceIndex`, so you index some type inside the slice.
Both, `key` and `identifier` somewhat imply to me, that only a single value can be returned from this function.

Reply:
Renamed them do `identifier`